### PR TITLE
Update to polkadot v0.9.37

### PR DESCRIPTION
**Pull request by Hounsette**
Main change: 

- Removal of wasmtime feature flag see [here](https://github.com/paritytech/substrate/pull/12684)
- Removal of instantiation for the pallet contract rpc see [here](https://github.com/paritytech/substrate/pull/12358)
- Usage of unstable feature for the pallet contract are activable/deactivable by a flag in the config [here](https://github.com/NodleCode/chain/pull/700/files#diff-bc781f5a3dd945b1c46c426a837c2bd87d1f19a31ba63080b52f1d615575951dR185)

- [x]  Run benchmark
- [x] Add fix for https://github.com/paritytech/substrate/pull/12661

### CODE REVIEW

File: runtimes/eden/src/pallets_util.rs [hunk 1]
```diff
@@ -182,4 +182,5 @@ impl pallet_contracts::Config for Runtime {
 	type MaxCodeLen = ConstU32<{ 128 * 1024 }>;
 	type RelaxedMaxCodeLen = ConstU32<{ 256 * 1024 }>;
 	type MaxStorageKeyLen = ConstU32<128>;
+	type UnsafeUnstableInterface = ConstBool<true>;
```
Comment (aliXsed) [inline runtimes/eden/src/pallets_util.rs:185]: It should be `false`, we haven't used the unstable interface

File: runtimes/eden/src/lib.rs [hunk 2]
```diff
@@ -267,7 +267,7 @@ sp_api::impl_runtime_apis! {
 		}
 	}
 
-	impl pallet_contracts_rpc_runtime_api::ContractsApi<Block, AccountId, Balance, BlockNumber, Hash>
+	impl pallet_contracts::ContractsApi<Block, AccountId, Balance, BlockNumber, Hash>
```
Comment (aliXsed) [inline runtimes/eden/src/lib.rs:270]: Without having an rpc_runtime_api, I wonder if `ContractsApi` exists in pallet_contracts or not?

Comment (Hounsette) [reply runtimes/eden/src/lib.rs:270]: it has been moved to the pallet_contract itself. [See](https://docs.rs/pallet-contracts/latest/pallet_contracts/trait.ContractsApi.html)

**Comment by Hounsette** [comment]
I force pushed to remove commit that I used to test the CI

**Comment by aliXsed** [comment]
@simonsso @Hounsette I have prepared a fix for the last item in the description of this pr. But I would rather make it a separate PR. So we only need the benchmarks for this one.

**Comment by Hounsette** [comment]
> @simonsso @Hounsette I have prepared a fix for the last item in the description of this pr. But I would rather make it a separate PR. So we only need the benchmarks for this one.

Seems good to me, we can separate the PR

File: node/src/command.rs [hunk 3]
```diff
@@ -246,20 +250,29 @@ pub fn run() -> Result<()> {
...
+				Ok((
+					cmd.run::<Block, HostFunctionsOf<TemplateRuntimeExecutor>>(),
+					task_manager,
+				))
+			})
 		}
+		#[cfg(not(feature = "try-runtime"))]
+		Some(Subcommand::TryRuntime) => Err("Try-runtime was not enabled when building the node. \
+			You can enable it with `--features try-runtime`."
+			.into()),
```
Comment (simonsso) [inline node/src/command.rs:275]: smile.

**Review by simonsso — DISMISSED** [review]

**Review by aliXsed — APPROVED** [review]
Thanks @Hounsette and @simonsso

**Pull request merged**